File: 950. Reveal Cards In Increasing Order/deckRevealedIncreasing.js
/**
 * @param {number[]} deck
 * @return {number[]}
 */
const deckRevealedIncreasing = (deck) => {
  deck.sort((a, b) => a - b);
  const l = deck.length;
  const res = Array(l).fill(0);
  const q = [...Array(l).keys()];

  for (let i = 0; i < l; i++) {
    let index = q.shift();
    res[index] = deck[i];

    if (q.length) {
      q.push(q.shift());
    }
  }

  return res;
};

const deck = [17, 13, 11, 2, 3, 5, 7];
console.log('deckRevealedIncreasing: ', deckRevealedIncreasing(deck));

File: 950. Reveal Cards In Increasing Order/deckRevealedIncreasing.py
from typing import List
from collections import deque

class Solution:
  def deckRevealedIncreasing(self, deck: List[int]) -> List[int]:
    res = [0] * len(deck)
    deck.sort()
    q = deque(range(len(deck)))
    
    for n in deck:
      i = q.popleft()
      res[i] = n
      
      if q:
        q.append(q.popleft())
      
    return res


deck = [17, 13, 11, 2, 3, 5, 7]
solution = Solution()
print('deckRevealedIncreasing: ', solution.deckRevealedIncreasing(deck))
